Sources/PitchSpeller/Data Structures/UndirectedEdge.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  UndirectedEdge.swift
+//  PitchSpeller
+//
+//  Created by Benjamin Wetherfield on 7/1/18.
+//
+
+public struct UndirectedEdge<Node, Num>: EdgeLike {
+    
+    typealias Weight = Num
+    typealias Pair = UnorderedPair<Node>
+    
+    var weight: Weight
+    let nodes: Pair
+    
+    init (_ source: Node, _ destination: Node, _ weight: Weight) {
```

Comment:
I don't always do this, but I think I should:

I like to call a higher-level initializer from a lower-level one:

```Swift
init (_ source: Node, _ destination: Node, _ weight: Weight) {
    self.init(Pair(source.destination), weight)
}
```

Reply:
I think this just chisels away at any potential code duplication, therefore potential double-maintainance scenarios.

Reply:
Ah cool!